PHPCompatibility/Sniffs/Syntax/NewInterpolatedStringDereferencingSniff.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2009-2020 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Sniffs\Syntax;
+
+use PHPCompatibility\Sniff;
+use PHPCSUtils\Utils\TextStrings;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Detect dereferencing of interpolated strings.
+ *
+ * PHP version 8.0
+ *
+ * @link https://wiki.php.net/rfc/variable_syntax_tweaks#interpolated_and_non-interpolated_strings
+ * @link https://www.php.net/manual/en/language.types.string.php#language.types.string.parsing
+ *
+ * @since 10.0.0
+ */
+class NewInterpolatedStringDereferencingSniff extends Sniff
+{
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @since 10.0.0
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            \T_DOUBLE_QUOTED_STRING,
+        ];
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 10.0.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        // PHP 8.0 supports dereferencing interpolated strings
+        if ($this->supportsBelow('7.4') === false) {
+            return $stackPtr;
```

Comment:
```suggestion
            return;
```

Returning the current stack pointer has no effect.